x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeFormatProcessor.java

```diff
@@ -36,22 +49,379 @@
         {"F", "S"},
         {"z", "X"}
     };
-    private final Formatter formatter;
+    
+    private static class ToCharFormatter {
```

Comment:
Will move this out into a separate file.

Reply:
Since it sounds like the PR is not finished, you can mark it as draft until you get all the changes in and make it final once you are done with it so folks can actually start reviewing it.